### PR TITLE
Feature#parsers refactoring

### PR DESCRIPTION
**Pull request by raphael-melo**
Allowing parsers to be set independently from the main code.

### CODE REVIEW

File: hkparser.js [hunk 1]
```diff
@@ -4,26 +4,27 @@
...
+const Context = HK.Context;
+const Reference = HK.Reference;
+const LAMBDA = HK.Constants.LAMBDA;
 
 const HYPERKNOWLEDGE_URIS = new Set();
 
 HYPERKNOWLEDGE_URIS.add(HKUris.HAS_PARENT_URI);
 HYPERKNOWLEDGE_URIS.add(HKUris.REFERENCES_URI);
 HYPERKNOWLEDGE_URIS.add(HKUris.USES_CONNECTOR_URI);
+HYPERKNOWLEDGE_URIS.add(HKUris.REFERENCED_BY_URI);
```
Comment (eltonfss) [inline hkparser.js:27]: REFERENCED_BY_URI  has been previously deprecated. Is the any specific need for adding it back to the lib?

File: parser.js [hunk 2]
```diff
@@ -112,11 +112,19 @@ function parseGraph(graph, options)
...
 	let blankNodesMap = {};
 
-	let owlParser = new OWLParser(entities, options);
-
-	let owlTimeParser = new OWLTimeParser(entities, options);
-
-	let hkParser = new HKParser(entities, blankNodesMap, onlyHK);
+  // instantiate new parsers
+  const parsers = [];
+  registeredParsers.forEach(parser => {
```
Comment (eltonfss) [inline parser.js:117]: Nice!

**Review by eltonfss — CHANGES_REQUESTED** [review]
- Where the indentation changes in "{" intentional?
- REFERENCED_BY_URI has been previously deprecated. Is the any specific need for adding it back to the lib?

File: hkparser.js (hunk 1 above)
Comment (raphael-melo) [reply hkparser.js:27]: Not really, I must have worked on an earlier commit. I'll remove it.

**Comment by raphael-melo** [comment]
> * Where the indentation changes in "{" intentional?

Changed to new lines. 

> * REFERENCED_BY_URI has been previously deprecated. Is the any specific need for adding it back to the lib?

**Review by eltonfss — APPROVED** [review]

**Pull request merged**